packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+        connector: this,
+        logLevel: this.options.logLevel,
+      };
+      const endpoint = new GetPrometheusExporterMetricsEndpointV1(opts);
+      endpoints.push(endpoint);
+    }
+    this.endpoints = endpoints;
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ledger-connector-iroha`;
+  }
+
+  public async getConsensusAlgorithmFamily(): Promise<
+    ConsensusAlgorithmFamily
+  > {
+    return ConsensusAlgorithmFamily.Authority;
+  }
+  public async hasTransactionFinality(): Promise<boolean> {
+    const currentConsensusAlgorithmFamily = await this.getConsensusAlgorithmFamily();
+
+    return consensusHasTransactionFinality(currentConsensusAlgorithmFamily);
+  }
+  //...  spread
+  //pass in the params
+  public async transact(
+    req: RunTransactionRequest, //string + array<any>
+  ): Promise<RunTransactionResponse> {
+    const { baseConfig } = req;
+    if (
+      !baseConfig ||
+      !baseConfig.privKey ||
+      !baseConfig.creatorAccountId ||
+      !baseConfig.irohaHost ||
+      !baseConfig.irohaPort ||
+      !baseConfig.quorum ||
+      !baseConfig.timeoutLimit
+    ) {
+      console.error("No data here!");
+      return { transactionReceipt: "Some fields in baseConfig is undefined" };
+    }
+    const irohaHostPort = `${baseConfig.irohaHost}:${baseConfig.irohaPort}`;
+    const adminPriv =
+      "f101537e319568c765b2cc89698325604991dca57b9716b58016b253506cab70";
+
+    const commandService = new CommandService(
+      irohaHostPort,
+      //TODO:do something in the production environment
+      grpc.credentials.createInsecure(),
+    );
+    const queryService = new QueryService(
+      irohaHostPort,
+      grpc.credentials.createInsecure(),
+    );
+    const commandOptions = {
+      privateKeys: baseConfig.privKey, //need an array of keys for command
+      creatorAccountId: baseConfig.creatorAccountId,
+      quorum: baseConfig.quorum,
+      commandService: commandService,
+      timeoutLimit: baseConfig.timeoutLimit,
+    };
+    const commandOptions2 = {
+      privateKeys: [adminPriv],
+      creatorAccountId: "admin@test",
+      quorum: 2,
+      commandService: commandService,
+      timeoutLimit: 5000,
+    };
+    const queryOptions = {
+      privateKey: baseConfig.privKey[0], //only need 1 key for query
+      creatorAccountId: baseConfig.creatorAccountId as string,
+      queryService: queryService,
+      timeoutLimit: baseConfig.timeoutLimit,
+    };
+    console.log(req);
+
+    switch (req.commandName) {
+      case IrohaCommand.CreateAccount: {
+        try {
+          const response = await commands.createAccount(commandOptions, {
+            accountName: req.params[0],
+            domainId: req.params[1],
+            publicKey: req.params[2],
+          });
+          return { transactionReceipt: response };
+        } catch (err) {
+          throw new Error(err);
```

Comment:
@hxlaf use the `run-time-error` module to re-throw. Standard Error class does not work like this (all information about the error could be lost)

(this applies to all instances of you using `throw` this way)

Reply:
I would assume that it also applied to cases like this?
![image](https://user-images.githubusercontent.com/10954157/129978338-e1c5da58-5b35-422a-bf38-9b5242ffee34.png)

Reply:
Applied `throw new RuntimeError` to all `throw` instances via the `run-time-error` module. Thank you!